metis-common/metis-common-utils/src/main/java/eu/europeana/metis/utils/GeoUriWGS84Parser.java

```diff
@@ -0,0 +1,214 @@
+        throw new BadContentException("Invalid geo uri 'u' parameter position");
+      }
+    }
+    //Validate value of crs
+    if (!CRS_WGS_84.equalsIgnoreCase(crs)) {
+      throw new BadContentException(String.format("Crs parameter value is not %s", CRS_WGS_84));
+    }
+
+    //Finally, check the coordinates part and validate
+    return getGeoCoordinates(geoUriParts[0]);
+  }
+
+  /**
+   * Generate a geo coordinates from a geoUriPart string.
+   * <p>The provided string is validated against:
+   *   <ul>
+   *     <li>the total coordinates available</li>
+   *     <li>the validity of each number and its range</li>
+   *     <li>the convertibility to a {@link Double}</li>
+   *   </ul>
+   *   The decimal points are also truncated up to a maximum allowed.
+   * </p>
+   *
+   * @param geoUriPart the string that should contain the coordinates
+   * @return the geo coordinates
+   * @throws BadContentException if the geo coordinates were not valid
+   */
+  private static GeoCoordinates getGeoCoordinates(String geoUriPart) throws BadContentException {
+    final String[] coordinates = geoUriPart.split(",");
+    if (coordinates.length < 2 || coordinates.length > MAX_NUMBER_COORDINATES) {
+      throw new BadContentException("Coordinates are not of valid length");
+    }
+    final Matcher latitudeMatcher = latitudePattern.matcher(coordinates[0]);
+    final Matcher longitudeMatcher = longitudePattern.matcher(coordinates[1]);
+    final GeoCoordinates geoCoordinates;
+    if (latitudeMatcher.matches() && longitudeMatcher.matches()) {
+      Double altitude = null;
+      if (coordinates.length == MAX_NUMBER_COORDINATES) {
+        final Matcher altitudeMatcher = altitudePattern.matcher(coordinates[2]);
+        if (altitudeMatcher.matches()) {
+          altitude = Double.parseDouble(truncateDecimalPoints(altitudeMatcher.group(0)));
+        }
+      }
+      geoCoordinates = new GeoCoordinates(
+          Double.parseDouble(truncateDecimalPoints(latitudeMatcher.group(0))),
+          Double.parseDouble(truncateDecimalPoints(longitudeMatcher.group(0))), altitude);
+    } else {
+      throw new BadContentException("Coordinates are invalid");
+    }
+    return geoCoordinates;
+  }
+
+  private static String truncateDecimalPoints(String decimalNumber) {
+    final String[] decimalNumberParts = decimalNumber.split("\\.");
+    final StringBuilder decimalNumberTruncated = new StringBuilder();
+    if (decimalNumberParts.length >= 1) {
+      decimalNumberTruncated.append(decimalNumberParts[0]);
+    }
+    if (decimalNumberParts.length > 1) {
+      decimalNumberTruncated.append(".");
+      decimalNumberTruncated.append(decimalNumberParts[1], 0,
+          Math.min(decimalNumberParts[1].length(), MAX_DECIMAL_POINTS_TO_KEEP));
+    }
+    return decimalNumberTruncated.toString();
+  }
+
+  /**
+   * Class containing geo coordinates (latitude, longitude)
+   */
+  public static class GeoCoordinates {
+
+    private final Double latitude;
```

Comment:
Hmmm might as well keep all with `Double` class to make it consistent. But since latitude and longitude cannot be null, perhaps we could add a check in the constructor to ensure they are not null?
Unless these checks are done everywhere these values are used, then just keep it as it is

Reply:
I thought about that, but since it's public I thought perhaps it could be used elsewhere.
The regexes guarantee their non-nullity essentially.
I'll leave as such then, but in general either way would be acceptable!